### PR TITLE
Migrate most of the SQL tests to the logic test framework.

### PR DESCRIPTION
**Pull request by petermattis**

### CODE REVIEW

**Comment by MBerhault** [comment]
LGTM. I haven't gone through it line-by-line, I'll assume you did.

**Comment by petermattis** [comment]
@MBerhault There were some minor changes to the insertion tests, but mostly this was just straightforward migration.

File: sql/logic_test.go [hunk 1]
```diff
@@ -201,28 +201,27 @@ func (t logicTest) run(path string) {
...
-				}
-				query.expectedHash = m[2]
-			} else {
-				for {
-					results := strings.Fields(s.Text())
-					if len(results) == 0 {
-						break
+			if s.Scan() {
+				if m := resultsRE.FindStringSubmatch(s.Text()); len(m) == 3 {
+					var err error
```
Comment (vivekmenezes) [inline sql/logic_test.go]: Why the magic number 3? Also for m[2], it will be nice to name '2'

Comment (petermattis) [inline sql/logic_test.go]: That's the number of capturing groups + 1. Might be better to make this test `m != nil`, though.

File: sql/driver/driver_test.go [hunk 2]
```diff
@@ -136,20 +136,18 @@ func verifyResults(expectedResults resultSlice, actualResults resultSlice) error
 	return nil
 }
 
-func TestTruncateTable(t *testing.T) {
+func TestPlaceholders(t *testing.T) {
```
Comment (tamird) [inline sql/driver/driver_test.go]: :+1:

File: sql/logic_test.go [hunk 3]
```diff
@@ -201,28 +201,27 @@ func (t logicTest) run(path string) {
 			// blank line or a line of the form "xx values hashing to yyy". The
 			// latter format is used by sqllogictest when a large number of results
 			// match the query.
-			if !s.Scan() {
-				t.Fatalf("%d: expected query results line\n", s.line)
```
Comment (tamird) [inline sql/logic_test.go]: should this live on in an `else` clause?

**Comment by tamird** [comment]
LGTM also did not review each individual test case

**Comment by vivekmenezes** [comment]
LGTM

File: sql/logic_test.go (hunk 3 above)
Comment (petermattis) [inline sql/logic_test.go]: This was only occurring at the end of a file. Without the fatal we'll determine the query has no results which seems reasonable.

File: sql/logic_test.go (hunk 1 above)
Comment (petermattis) [inline sql/logic_test.go]: Changed to `m != nil`. When using regexp matches you usually assume you know the indexes of the capturing groups.

**Pull request merged**